chaco/tools/tests/test_range_selection.py

```diff
@@ -75,3 +76,28 @@ def test_selection_no_warning(self):
         tool.selection = (1.5, 3.5)
         tool.selection = [1.0, 2.0]
         tool.selection = None
+
+    # regression test for enthought/chaco#597
+    @unittest.mock.patch('chaco.tools.range_selection.RangeSelection.deselect')
+    def test_notifiers_connected(self, mocked_deselect):
+        plot_data = ArrayPlotData()
+        arr = np.arange(4.0)
+        plot_data.set_data("x", arr)
+        plot_data.set_data("y", arr)
+
+        plot = Plot(plot_data)
+
+        renderer = plot.plot(("x", "y"))[0]
+        renderer.bounds = [10, 20]
+        tool = RangeSelection(renderer)
+        renderer.tools.append(tool)
+
+        # attempt to trigger change handler for the index_mapper trait on the
+        # RangeSelection tool's plot
+        # assign a new mapper with same attrs
+        renderer.index_mapper = LinearMapper(
+            range=renderer.index_mapper.range,
+            stretch_data=renderer.index_mapper.stretch_data
+        )
+
+        mocked_deselect.assert_called_once()
```

Comment:
This test fails, and the error is still observed if the fix in the file above is replaced with:
```
@observe("component")
def _component_updated(self, event):
    if event.old is not None:
        self.plot.observe(
            self.__mapper_changed, self.axis + "_mapper", remove=True
        )
    if event.new is not None:
        self.plot.observe(self.__mapper_changed, self.axis + "_mapper")
```

seems the timing at which notifiers are setup/called is different for the magic named methods vs decorating with observe?